File: app/src/main/java/eu/darken/bluemusic/bluetooth/core/FakeSpeakerDevice.java
package eu.darken.bluemusic.bluetooth.core;

import android.bluetooth.BluetoothClass;
import android.content.Context;
import android.os.Parcel;
import android.os.Parcelable;
import android.support.annotation.Nullable;

import eu.darken.bluemusic.R;
import eu.darken.bluemusic.main.core.audio.AudioStream;


public class FakeSpeakerDevice implements SourceDevice, Parcelable {
    public static final String ADDR = "self:speaker:main";
    private final String alias;

    public FakeSpeakerDevice(Context context) {
        alias = context.getString(R.string.label_device_speaker);
    }

    @Nullable
    @Override
    public BluetoothClass getBluetoothClass() {
        return null;
    }

    @Nullable
    @Override
    public String getName() {
        return android.os.Build.MODEL;
    }

    @Override
    public String getAddress() {
        return ADDR;
    }

    @Override
    public boolean setAlias(String newAlias) {
        return false;
    }

    @Nullable
    @Override
    public String getAlias() {
        return "Speaker";
    }

    @Override
    public AudioStream.Id getStreamId(AudioStream.Type type) {
        switch (type) {
            case MUSIC:
                return AudioStream.Id.STREAM_MUSIC;
            case CALL:
                return AudioStream.Id.STREAM_VOICE_CALL;
            case RINGTONE:
                return AudioStream.Id.STREAM_RINGTONE;
            default:
                throw new IllegalArgumentException("Unknown AudioStreamType: " + type);
        }
    }

    protected FakeSpeakerDevice(Parcel in) {
        alias = (String) in.readValue(getClass().getClassLoader());
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel parcel, int i) {
        parcel.writeValue(alias);
    }

    public static final Creator<FakeSpeakerDevice> CREATOR = new Creator<FakeSpeakerDevice>() {
        @Override
        public FakeSpeakerDevice createFromParcel(Parcel in) {
            return new FakeSpeakerDevice(in);
        }

        @Override
        public FakeSpeakerDevice[] newArray(int size) {
            return new FakeSpeakerDevice[size];
        }
    };

    @Override
    public String toString() {
        return "FakeSpeakerDevice()";
    }
}

File: app/src/main/java/eu/darken/bluemusic/bluetooth/core/speaker/FakeSpeakerDevice.kt
package eu.darken.bluemusic.bluetooth.core.speaker

import android.os.Parcelable
import eu.darken.bluemusic.bluetooth.core.SourceDevice
import eu.darken.bluemusic.monitor.core.audio.AudioStream
import kotlinx.parcelize.Parcelize

@Parcelize
class FakeSpeakerDevice(
    override val label: String,
    override val isActive: Boolean,
) : SourceDevice, Parcelable {

    override val address: String
        get() = ADDRESS

    override val deviceType: SourceDevice.Type
        get() = SourceDevice.Type.PHONE_SPEAKER

    override fun getStreamId(type: AudioStream.Type): AudioStream.Id = when (type) {
        AudioStream.Type.MUSIC -> AudioStream.Id.STREAM_MUSIC
        AudioStream.Type.CALL -> AudioStream.Id.STREAM_VOICE_CALL
        AudioStream.Type.RINGTONE -> AudioStream.Id.STREAM_RINGTONE
        AudioStream.Type.NOTIFICATION -> AudioStream.Id.STREAM_NOTIFICATION
        AudioStream.Type.ALARM -> AudioStream.Id.STREAM_ALARM
    }

    override fun toString(): String {
        return "FakeSpeakerDevice(isActive=$isActive)"
    }

    companion object {
        private const val ADDRESS = "self:speaker:main"
    }
}